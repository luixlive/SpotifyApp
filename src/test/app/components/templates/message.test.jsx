import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import initialState from './../../../test_utils/initial_state';
import {
  message as componentProps,
} from './../../../test_utils/components_props';
import Message, {
  PureMessage,
} from './../../../../app/components/templates/message';

describe('App Components Templates - Message', () => {
  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered = renderer.create(<PureMessage {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.deviceMobile = true;
      const rendered = renderer.create(<PureMessage {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders custom title', () => {
      props.title = 'title';
      const rendered = renderer.create(<PureMessage {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <Message title={componentProps.title}>
            {componentProps.children}
          </Message>
        </Provider>));
    });

    it('renders', () => {
      expect(wrapper.find(PureMessage).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureMessage).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
