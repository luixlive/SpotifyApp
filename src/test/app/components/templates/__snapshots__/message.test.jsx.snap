// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components Templates - Message Snapshots renders custom title 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    title
  </h1>
  <h2
    className="ui header"
  >
    <div />
  </h2>
</div>
`;

exports[`App Components Templates - Message Snapshots renders mobile 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "2em",
        "marginTop": "1em",
      }
    }
  >
    
  </h1>
  <h3
    className="ui header"
  >
    <div />
  </h3>
</div>
`;

exports[`App Components Templates - Message Snapshots renders no mobile 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    
  </h1>
  <h2
    className="ui header"
  >
    <div />
  </h2>
</div>
`;
